From 76b69b9410624910ddf63644b1e9891b5d1b31f0 Mon Sep 17 00:00:00 2001
From: tsujan <tsujan2000@gmail.com>
Date: Tue, 21 Jun 2022 01:16:29 +0430
Subject: [PATCH] Fixed multi-screen screenshot of window and its decoration
 (#305)

Previously, a wrong position was given to `QScreen::grabWindow()` with the root window.

Fixes https://github.com/lxqt/lxqt/issues/2251
---
 src/core/core.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/core/core.cpp b/src/core/core.cpp
index 7f51b80..1e359b5 100644
--- a/src/core/core.cpp
+++ b/src/core/core.cpp
@@ -318,7 +318,14 @@ void Core::getActiveWindow() // called only with window screenshots
     // The offscreen part of the window will appear as a black area in the screenshot.
     // Until a better method is found, the offscreen area is ignored here.
     QRect r = screen->virtualGeometry().intersected(geometry);
-    *_pixelMap = screen->grabWindow(0, r.x(), r.y(), r.width(), r.height());
+
+    // the window positon should be calculated relative to the screen
+    *_pixelMap = screen->grabWindow(0,
+                                    r.x() - screen->geometry().x(),
+                                    r.y() - screen->geometry().y(),
+                                    r.width(),
+                                    r.height());
+
     grabCursor(geometry.x(), geometry.y());
 }
 
